src/util-lua-common.c

```diff
@@ -605,69 +612,93 @@ static int LuaCallbackRuleActionPushToStackFromPacketAlert(
+ *  otherwise pulled from Signature in lua registry (for match scripts)
  */
 static int LuaCallbackRuleAction(lua_State *luastate)
 {
+    const Signature *s = NULL;
     const PacketAlert *pa = LuaStateGetPacketAlert(luastate);
-    if (pa == NULL)
-        return LuaCallbackError(luastate, "internal error: no packet");
-
-    return LuaCallbackRuleActionPushToStackFromPacketAlert(luastate, pa);
+    if (pa != NULL) {
+        s = pa->s;
+    } else {
+        s = LuaStateGetSignature(luastate);
+        if (s == NULL)
+            return LuaCallbackError(luastate, "internal error: no packet alert or signature");
+    }
+    return LuaCallbackRuleActionPushToStackFromSignature(luastate, s);
 }
 
 /** \internal
- *  \brief fill lua stack with alert info
+ *  \brief fill lua stack with signature info
  *  \param luastate the lua state
- *  \param pa pointer to packet alert struct
+ *  \param s pointer to signature struct
  *  \retval cnt number of data items placed on the stack
  *
  *  Places: msg (string)
  */
-static int LuaCallbackRuleMsgPushToStackFromPacketAlert(lua_State *luastate, const PacketAlert *pa)
+static int LuaCallbackRuleMsgPushToStackFromSignature(lua_State *luastate, const Signature *s)
 {
-    lua_pushstring (luastate, pa->s->msg);
+    lua_pushstring(luastate, s->msg);
     return 1;
 }
 
 /** \internal
  *  \brief Wrapper for getting tuple info into a lua script
  *  \retval cnt number of items placed on the stack
+ *
+ *  Info is pulled from PacketAlert if it exists in lua registry (true for logging scripts)
+ *  otherwise pulled from Signature in lua registry (for match scripts)
  */
 static int LuaCallbackRuleMsg(lua_State *luastate)
 {
+    const Signature *s = NULL;
     const PacketAlert *pa = LuaStateGetPacketAlert(luastate);
-    if (pa == NULL)
-        return LuaCallbackError(luastate, "internal error: no packet");
-
-    return LuaCallbackRuleMsgPushToStackFromPacketAlert(luastate, pa);
+    if (pa != NULL) {
+        s = pa->s;
+    } else {
+        s = LuaStateGetSignature(luastate);
+        if (s == NULL)
+            return LuaCallbackError(luastate, "internal error: no packet alert or signature");
+    }
+    return LuaCallbackRuleMsgPushToStackFromSignature(luastate, s);
 }
 
 /** \internal
- *  \brief fill lua stack with alert info
+ *  \brief fill lua stack with signature info
  *  \param luastate the lua state
- *  \param pa pointer to packet alert struct
+ *  \param s pointer to signature struct
  *  \retval cnt number of data items placed on the stack
  *
  *  Places: class (string), prio (number)
  */
-static int LuaCallbackRuleClassPushToStackFromPacketAlert(lua_State *luastate, const PacketAlert *pa)
+static int LuaCallbackRuleClassPushToStackFromSignature(lua_State *luastate, const Signature *s)
 {
-    lua_pushstring (luastate, pa->s->class_msg);
-    lua_pushnumber (luastate, pa->s->prio);
+    lua_pushstring(luastate, s->class_msg);
+    lua_pushnumber(luastate, s->prio);
```

Comment:
I think it should be `lua_pushinteger` here, to prevent it from being considered a float in newer Lua versions...